torch/distributed/device_mesh.py

```diff
@@ -156,20 +108,18 @@ def create_flatten_mesh(
         ) -> "DeviceMesh":
             root_mesh = _mesh_resources.get_root_mesh(device_mesh)
 
-            flatten_dims_in_root = [
-                not_none(root_mesh.mesh_dim_names).index(flatten_mesh_dim_name)
-                for flatten_mesh_dim_name in not_none(device_mesh.mesh_dim_names)
-            ]
-
             if not mesh_dim_name:
                 mesh_dim_name = "_".join(not_none(device_mesh.mesh_dim_names))
 
+            # Flatten a 1D device mesh into its original mesh_dim_name will return itself.
+            if device_mesh.ndim == 1 and mesh_dim_name in not_none(
+                device_mesh.mesh_dim_names
+            ):
+                return device_mesh
+
             # Check whether the mesh_dim_name for flattened mesh is valid.
-            self.flatten_name_to_root_dims.setdefault(root_mesh, {})
-            invalid_dim_names = chain(
-                list(not_none(root_mesh.mesh_dim_names)),
-                *self.flatten_name_to_root_dims[root_mesh].keys(),
-            )
+            self.flatten_name_to_root_layout.setdefault(root_mesh, {})
+            invalid_dim_names = not_none(root_mesh.mesh_dim_names)
```

Comment:
Why do you get to eliminate `flatten_name_to_root_dims`?

Reply:
great questions, previously when do slicing after flatten, we need to combine dims from `flatten_name_to_root_dims` and dims from mesh to build the submesh. Now we directly store the layout, this is not needed?